### PR TITLE
cmd/swarm: handle SIGTERM unix signal for clean exit

### PR DESCRIPTION
**Pull request by zelig**
docker container swarm instances used SIGTERM to shutdown
so swarm nodes on the testnet never exited cleanly
therefore they didnt save the syncstate and peers
so everytime they restarted, nodes started syncing from scratch.
as they had the chunks only hashes were spuriously exchanged but that still
likely caused a lot of unnecessary traffic

### CODE REVIEW

**Comment by homotopycolimit** [comment]
as the admin running the swarm testnet cluster, I fully support this PR.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 6cf07897b76a0517a824098113973b910fd1b319
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 53aced33267b634cfade7ddd2ff83530448519c5
  * Commits must be prefixed with the package(s) they modify

 * Commit: 43a9dcd0195b20beff571754fa9567dff50a7470
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 53aced33267b634cfade7ddd2ff83530448519c5
  * Commits must be prefixed with the package(s) they modify

 * Commit: 43a9dcd0195b20beff571754fa9567dff50a7470
  * Commits must be prefixed with the package(s) they modify

 * Commit: ba99c034c962e57b827070e71a4221bcf6211833
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 53aced33267b634cfade7ddd2ff83530448519c5
  * Commits must be prefixed with the package(s) they modify

 * Commit: 43a9dcd0195b20beff571754fa9567dff50a7470
  * Commits must be prefixed with the package(s) they modify

 * Commit: ba99c034c962e57b827070e71a4221bcf6211833
  * Commits must be prefixed with the package(s) they modify

 * Commit: a0c8e51a52af560aaa9f7b392096c9915f3dd113
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 53aced33267b634cfade7ddd2ff83530448519c5
  * Commits must be prefixed with the package(s) they modify

 * Commit: 43a9dcd0195b20beff571754fa9567dff50a7470
  * Commits must be prefixed with the package(s) they modify

 * Commit: ba99c034c962e57b827070e71a4221bcf6211833
  * Commits must be prefixed with the package(s) they modify

 * Commit: 9cbdb7c7f2aee7c839cee110ce1c274c0abde62d
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 53aced33267b634cfade7ddd2ff83530448519c5
  * Commits must be prefixed with the package(s) they modify

 * Commit: 43a9dcd0195b20beff571754fa9567dff50a7470
  * Commits must be prefixed with the package(s) they modify

 * Commit: ba99c034c962e57b827070e71a4221bcf6211833
  * Commits must be prefixed with the package(s) they modify

 * Commit: 9cbdb7c7f2aee7c839cee110ce1c274c0abde62d
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: cmd/swarm/main.go [hunk 1]
```diff
@@ -224,6 +226,14 @@ func bzzd(ctx *cli.Context) error {
...
 	registerBzzService(ctx, stack)
 	utils.StartNode(stack)
+	go func() {
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, syscall.SIGTERM)
+		defer signal.Stop(sigc)
+		<-sigc
+		glog.V(logger.Info).Infoln("Got sigterm, shutting down...")
+		stack.Stop()
+	}()
```
Comment (fjl) [inline cmd/swarm/main.go]: utils.StartNode does a very similar thing. Please consider adding support for it there instead of repeating the same code here.

Comment (homotopycolimit) [inline cmd/swarm/main.go]: I think I just did that. Can you rereview? :)

**Comment by karalabe** [comment]
@zelig Builds fails.

**Comment by homotopycolimit** [comment]
@karalabe sorry. that was my fault.

**Comment by karalabe** [comment]
@homotopycolimit Thanks, will merge if green.

**Comment by karalabe** [comment]
@zelig @homotopycolimit Ok, so now you force pushed again, resetting all progress and work.

**Comment by homotopycolimit** [comment]
@karalabe sorry. Did I do something wrong?
I thought I was supposed to use force push so that there is only one commit.
sincere apologies if I caused anyone more work.

What is the correct way to amend a commit?

**Comment by karalabe** [comment]
@zelig @homotopycolimit Both of you are completely unaware that the other is force pushing and both of you are force pushing out each other's work.

**Comment by homotopycolimit** [comment]
Hey everyone.
Sorry it was all my fault.
I was pushing to Viktor's branch when I shouldn't have been.

**Pull request merged**